core/src/main/java/org/apache/iceberg/hadoop/HadoopCatalog.java

```diff
@@ -200,7 +211,110 @@ public void renameTable(TableIdentifier from, TableIdentifier to) {
   }
 
   @Override
-  public void close() throws IOException {
+  public void createNamespace(Namespace namespace, Map<String, String> meta) {
+    Preconditions.checkArgument(!namespace.isEmpty(),
+        "Cannot create namespace with invalid name: %s", namespace);
+    if (meta.size() > 0) {
+      LOG.warn("Hadoop Catalog not support metadata {} on namespace: {}", meta, namespace);
```

Comment:
This should fail instead of ignoring the metadata. Otherwise this is dropping information.